app.py

```diff
@@ -242,6 +243,25 @@ def __add_write_node_commands(self, context=None):
                     context=context,
                 )
             )
-            
-            
 
+        # Show the convert actions in the Menu if configured to do so
+        if self.get_setting('show_convert_actions'):
+
+            self.engine.register_command(
+                "Convert from Shotgun Write Nodes",
```

Comment:
i think we should tweak the naming - i think we can make it even more self explanatory by naming these:

```
Convert SG Write Nodes to Write Nodes
Convert Write Nodes to SG Write Nodes
```

I think we also need a pop up box saying "this will convert all the nodes in the scene bla bla" - ok to proceed? for both commands.

Reply:
Do we want to avoid the pop-up box thing, as we decided against having one for the convert back method, to warn that it's not always going to return you to the same state as before? Because it would be annoying for them to press each time.

Perhaps we can name the buttons to be:

```
Convert All SG Write Nodes to Write Nodes
Convert All Write Nodes to SG Write Nodes
```

Otherwise, I guess the ideal situation would be to have a don't notify me again button, but we don't do that in any of our other apps really.

Reply:
Yes, for this sort of bulk operation, i think we should ask for confirmation and include [an ellipsis](https://developer.apple.com/design/human-interface-guidelines/macos/menus/menu-anatomy/):

```
Convert All SG Write Nodes to Write Nodes...
Convert All Write Nodes to SG Write Nodes...
```

This is not a terribly high frequency operation, make sure that hitting enter as the modal "are you sure" dialog is shown results in an "OK" and hitting escape equals cancel and we should be good!

Reply:
<img width="218" alt="colourwheel_v001_nk__modified__-_nuke" src="https://user-images.githubusercontent.com/3777228/42454267-f5821b82-8386-11e8-864c-2b0e925d7815.png">
<img width="223" alt="colourwheel_v001_nk__modified__-_nuke" src="https://user-images.githubusercontent.com/3777228/42454289-0628443e-8387-11e8-95c7-766641e2a9ab.png">

Added warning message boxes, to the existing public methods for converting. To enable these warning message boxes you must pass a kwarg of show_warning=True. Defaults to False so shouldn't break or change existing behaviour.
Imports of Qt were kept inside the convert methods so that the write node app didn't have a general dependacy on a GUI being present.

Reply:
This shouldn't be an outdated comment.

Reply:
<img width="216" alt="colourwheel_v001_nk__modified__-_nuke" src="https://user-images.githubusercontent.com/3777228/42456370-93a8975a-838c-11e8-8f11-548fa2b5f8df.png">
<img width="221" alt="colourwheel_v001_nk__modified__-_nuke" src="https://user-images.githubusercontent.com/3777228/42456386-9f1b1cde-838c-11e8-91cd-36cc8c7b59d6.png">

Updated the message boxes, with a few tweaks. (thanks @Masepack )
Do you think its good to call the convert back experimental? In my brief test, it was a bit unreliable and I wasn't promoting knobs either.